src/snapred/backend/data/LocalDataService.py

```
@@ -1258,3 +1259,97 @@ def writePixelMask(
         # At present, this method is just a wrapper for 'writeDiffCalWorkspaces':
         #   its existence allows for the separation of pixel-mask I/O from diffraction-calibration workspace I/O.
         self.writeDiffCalWorkspaces(path, filename, maskWorkspaceName=maskWorkspaceName)
+
+    ## LIVE-DATA SUPPORT METHODS
+
+    def hasLiveDataConnection(self) -> bool:
+        """For 'live data' methods: test if there is a listener connection to the instrument."""
+
+        # In addition to 'analysis.sns.gov', other nodes on the subnet should be OK as well.
+        #   So this check should also return True on those nodes.
+        # If this method returns True, then the `SNSLiveEventDataListener` should be able to function.
+
+        # Normalize to an actual "URL" and then strip off the protocol (not actually "http") and port:
+        #   `liveDataAddress` returns a string similar to "bl3-daq1.sns.gov:31415".
+
+        facility, instrument = Config["liveData.facility.name"], Config["liveData.instrument.name"]
+        hostname = urlparse(
+            "http://" + ConfigService.getFacility(facility).instrument(instrument).liveDataAddress()
+        ).hostname
+        status = True
+        try:
+            socket.gethostbyaddr(hostname)
+        except Exception:  # noqa: BLE001
+            # specifically:
```

print stacktrace?  I dont know if this swallows the error message.